### PR TITLE
Flags check

### PR DESCRIPTION
**Pull request by hallaroo**
First round of flags checking between v13 and v14. In my last PR @deepthi noted that some of the flags should not have been in the vtctld.md file.

I've removed the ones she noted and updated v13. And added all of the v13 flags to v14.

If the flags here can be reviewed that would be great. Then I'll move on to the next file. Thanks!

### CODE REVIEW

File: content/en/docs/14.0/reference/programs/vtctld.md [hunk 1]
```diff
@@ -27,253 +27,273 @@ vtctld \
...
 | -proxy_tablets | boolean | Setting this true will make vtctld proxy the tablet status instead of redirecting to them |
-| -purge_logs_interval | duration | how often try to remove old logs (default 1h0m0s) |
+| -purge_logs_interval | duration | How often to try to remove old logs (default 1h0m0s) |
 | -query-log-stream-handler | string | URL handler for streaming queries log (default "/debug/querylog") |
-| -querylog-filter-tag | string | string that must be present in the query as a comment for the query to be logged, works for both vtgate and vttablet |
-| -querylog-format | string | format for query logs ("text" or "json") (default "text") |
-| -queryserver-config-acl-exempt-acl | string | an acl that exempt from table acl checking (this acl is free to access any vitess tables). |
+| -querylog-filter-tag | string | String that must be present in the query as a comment for the query to be logged, works for both vtgate and vttablet |
+| -querylog-format | string | Gormat for query logs ("text" or "json") (default "text") |
+| -queryserver-config-acl-exempt-acl | string | An ACL that exempt from table ACL checking (this ACL is free to access any vitess tables). |
```
Comment (deepthi) [inline content/en/docs/14.0/reference/programs/vtctld.md]: Remove all `queryserver-` flags

File: content/en/docs/14.0/reference/programs/vtctld.md [hunk 2]
```diff
@@ -27,253 +27,273 @@ vtctld \
...
-| -onterm_timeout | duration | wait no more than this for OnTermSync handlers before stopping (default 10s) |
-| -opentsdb_uri | string | URI of opentsdb /api/put method |
-| -pid_file | string | If set, the process will write its pid to the named file, and delete it on graceful shutdown. |
-| -pool_hostname_resolve_interval | duration | if set force an update to all hostnames and reconnect if changed, defaults to 0 (disabled) |
-| -port | int | port for the server |
+| -mysql_clientcert_auth_method | string | Client-side authentication method to use. Supported values: mysql_clear_password, dialog (default "mysql_clear_password") |
+| -mysql_server_flush_delay | duration | Delay after which buffered response will flushed to client (default 100ms) |
+| -mysqlctl_client_protocol | string | The protocol to use to talk to the mysqlctl server (default "grpc") |
+| -mysqlctl_mycnf_template | string | Template file to use for generating the my.cnf file during server init |
+| -mysqlctl_socket | string | Socket file to use for remote mysqlctl actions (empty for local actions) |
```
Comment (deepthi) [inline content/en/docs/14.0/reference/programs/vtctld.md]: All these can be removed

File: content/en/docs/14.0/reference/programs/vtctld.md [hunk 3]
```diff
@@ -27,253 +27,273 @@ vtctld \
...
+| -s3_backup_server_side_encryption | string   | Server-side encryption algorithm (e.g., AES256, aws:kms, sse_c:/path/to/key/file) |
+| -s3_backup_storage_bucket | string   | S3 bucket to use for backups |
+| -s3_backup_storage_root | string   | Root prefix for all backup-related object names |
+| -s3_backup_tls_skip_verify_cert | boolean | Skip the 'certificate is valid' check for SSL connections |
+| -schema_change_check_interval | int | This value decides how often we check schema change dir, in seconds (default 60) |
+| -schema_change_controller | string | Schema change controller is responsible for finding schema changes and responding to schema change events |
+| -schema_change_dir | string | Directory contains schema changes for all keyspaces. Each keyspace has its own directory and schema changes are expected to live in '$KEYSPACE/input' dir. e.g. test_keyspace/input/*sql, each SQL file represents a schema change |
+| -schema_change_replicas_timeout | duration | How long to wait for replicas to receive the schema change (default 10s) |
```
Comment (deepthi) [inline content/en/docs/14.0/reference/programs/vtctld.md:151]: @rohit-nayak-ps @shlomi-noach are any of these `schema_change` options relevant to vtctld?
`schema_change_dir` looks like a vtcombo/vttestserver option, not sure about the other 3.

File: content/en/docs/14.0/reference/programs/vtctld.md [hunk 4]
```diff
@@ -27,253 +27,273 @@ vtctld \
...
+| -schema_swap_delay_between_errors | duration | Time to wait after a retryable error happened in the schema swap process (default 1m0s) |
+| -schema_swap_reparent_timeout | duration | Timeout to wait for replicas when doing reparent during schema swap (default 30s) |
+| -security_policy | string   | The name of a registered security policy to use for controlling access to URLs - empty means allow all for anyone (built-in policies: deny-all, read-only) |
+| -service_map | value    | Comma separated list of services to enable (or disable if prefixed with '-') Example: grpc-vtworker |
+| -serving_state_grace_period | duration | how long to pause after broadcasting health to vtgate, before enforcing a new serving state |
+| -shutdown_grace_period | float    | how long to wait (in seconds) for queries and transactions to complete during graceful shutdown. |
+| -sql-max-length-errors | int | Truncate queries in error logs to the given length (default unlimited) |
+| -sql-max-length-ui | int | Truncate queries in debug UIs to the given length (default 512) (default 512) |
```
Comment (deepthi) [inline content/en/docs/14.0/reference/programs/vtctld.md]: delete these `grace_period` flags

File: content/en/docs/14.0/reference/programs/vtctld.md [hunk 5]
```diff
@@ -27,253 +27,273 @@ vtctld \
...
+| -transaction_limit_by_principal | boolean | Include CallerID.principal when considering who the user is for the purpose of transaction limit (default true) |
+| -transaction_limit_by_subcomponent | boolean | Include CallerID.subcomponent when considering who the user is for the purpose of transaction limit |
+| -transaction_limit_by_username | boolean | Include VTGateCallerID.username when considering who the user is for the purpose of transaction limit (default true) |
+| -transaction_limit_per_user | float | Maximum number of transactions a single user is allowed to use at any time, represented as fraction of -transaction_cap (default 0.4) |
+| -transaction_shutdown_grace_period | int | How long to wait (in seconds) for transactions to complete during graceful shutdown |
+| -twopc_abandon_age | float | Time in seconds. Any unresolved transaction older than this time will be sent to the coordinator to be resolved |
+| -twopc_coordinator_address | string | Address of the (VTGate) process(es) that will be used to notify of abandoned transactions |
+| -twopc_enable | boolean | If the flag is on, 2pc is enabled. Other 2pc flags must be supplied |
```
Comment (deepthi) [inline content/en/docs/14.0/reference/programs/vtctld.md:218]: I believe these are all vttablet flags, @sougou can you confirm? If yes we can delete them from here.

File: content/en/docs/14.0/reference/programs/vtctld.md [hunk 6]
```diff
@@ -27,253 +27,273 @@ vtctld \
...
+| -vreplication_heartbeat_update_interval | int      | Frequency (in seconds, max 60) at which the time_updated column of a vreplication stream when idling (default 1) |
+| -vreplication_replica_lag_tolerance | duration | Replica lag threshold duration: once lag is below this we switch from copy phase to the replication (streaming) phase (default 1m0s) |
+| -vreplication_retry_delay | duration | Delay before retrying a failed binlog connection (default 5s) |
+| -vreplication_store_compressed_gtid | boolean | Store compressed gtids in the pos column of _vt.vreplication |
+| -vreplication_tablet_type | string   | Comma separated list of tablet types used as a source (default "PRIMARY,REPLICA") |
+| -vstream_dynamic_packet_size | boolean | Enable dynamic packet sizing for VReplication. This will adjust the packet size during replication to improve performance (default true) |
+| -vstream_packet_size | int      | Suggested packet size for VReplication streamer. This is used only as a recommendation. The actual packet size may be more or less than this amount (default 250000) |
```
Comment (deepthi) [inline content/en/docs/14.0/reference/programs/vtctld.md]: @rohit-nayak-ps are any of these `vreplication` flags actually relevant for vtctld?

File: content/en/docs/14.0/reference/programs/vtctld.md [hunk 7]
```diff
@@ -27,253 +27,273 @@ vtctld \
...
+| -xtrabackup_backup_flags | string   | Flags to pass to backup command. These should be space separated and will be added to the end of the command |
+| -xtrabackup_prepare_flags | string   | Flags to pass to prepare command. These should be space separated and will be added to the end of the command |
+| -xtrabackup_root_path | string   | Directory location of the xtrabackup and xbstream executables, e.g., /usr/bin |
+| -xtrabackup_stream_mode | string   | Which mode to use if streaming, valid values are tar and xbstream (default "tar") |
+| -xtrabackup_stripe_block_size | uint     | Size in bytes of each block that gets sent to a given stripe before rotating to the next stripe (default 102400) |
+| -xtrabackup_stripes | uint     | If greater than 0, use data striping across this many destination files to parallelize data transfer and decompression |
+| -xtrabackup_user | string   | User that xtrabackup will use to connect to the database server. This user must have all necessary privileges. For details, please refer to xtrabackup documentation |
```
Comment (deepthi) [inline content/en/docs/14.0/reference/programs/vtctld.md]: delete all `xtrabackup` flags.

File: content/en/docs/14.0/reference/programs/vtctld.md (hunk 3 above)
Comment (shlomi-noach) [reply content/en/docs/14.0/reference/programs/vtctld.md:151]: `schema_change_check_interval` is to be removed. Per https://github.com/vitessio/vitess/issues/6926#issuecomment-817619302 I could have done so for v13 (but didn't) -- I'll remove this shortly. I'll remove both code and tdocumentation. This is part of the initial Online DDL logic, where `vtctld` continuously probed for new migration requests in `topo`.

I am not familiar with `schema_change_controller`, `schema_change_dir` and `schema_change_replicas_timeout` - they're not part of Online DDL.

Comment (shlomi-noach) [reply content/en/docs/14.0/reference/programs/vtctld.md:151]: Hmmm actually `schema_change_check_interval` is used for other logic. I'm not familiar.

File: content/en/docs/14.0/reference/programs/vtctld.md (hunk 6 above)
Comment (rohit-nayak-ps) [reply content/en/docs/14.0/reference/programs/vtctld.md]: > @rohit-nayak-ps are any of these `vreplication` flags actually relevant for vtctld?

None of these are relevant for vtctld, only for vttablet.

File: content/en/docs/14.0/reference/programs/vtctld.md (hunk 4 above)
Comment (hallaroo) [reply content/en/docs/14.0/reference/programs/vtctld.md:192]: Just to check what about '-transaction_shutdown_grace_period' should that be removed too?

**Comment by deepthi** [comment]
@hallaroo any chance you are up for re-doing this PR? apologies for letting it get stale. the xtrabackup changes are still needed, and maybe others as well.

**Comment by hallaroo** [comment]
Hey sorry it took me so long to get back here. I am indeed up for re-doing this. I'll close this one out and get a fresh one open.

**Pull request closed without merging**